pyfr/integrators/writers.py

```diff
@@ -61,43 +63,106 @@ def _get_name_for_soln(self, etype, prank=None):
         return 'soln_{}_p{}'.format(etype, prank)
 
 
-class FileWriter(BaseWriter):
-    writer_name = 'pyfrs-file'
+class H5Writer(BaseWriter):
+    writer_name = 'pyfrs-h5-file'
 
     def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+        super(H5Writer, self).__init__(*args, **kwargs)
 
         # MPI info
         comm, rank, root = get_comm_rank_root()
 
         # Get the type and shape of each element in the partition
         etypes, shapes = self._system.ele_types, self._system.ele_shapes
-
         # Gather this information onto the root rank
         eleinfo = comm.gather(zip(etypes, shapes), root=root)
 
-        if rank == root:
-            self._mpi_rbufs = mpi_rbufs = []
-            self._mpi_rreqs = mpi_rreqs = []
-            self._mpi_names = mpi_names = []
-            self._loc_names = loc_names = []
-
-            for mrank, meleinfo in enumerate(eleinfo):
-                prank = self.rallocs.mprankmap[mrank]
-                for tag, (etype, dims) in enumerate(meleinfo):
-                    name = self._get_name_for_soln(etype, prank)
-
-                    if mrank == root:
-                        loc_names.append(name)
-                    else:
-                        rbuf = np.empty(dims, dtype=self.backend.fpdtype)
-                        rreq = comm.Recv_init(rbuf, mrank, tag)
-
-                        mpi_rbufs.append(rbuf)
-                        mpi_rreqs.append(rreq)
-                        mpi_names.append(name)
+        # Deciding if parallel
+        parallel = h5py.get_config().mpi
+        parallel &= h5py.version.version_tuple[0] >= 2
+        parallel &= h5py.version.version_tuple[1] >= 5
+        parallel &= not self.cfg.getbool('soln-output', 'serial-h5', False)
+
+        if(parallel):
+            self._write = self.__writeParallel
+        else:
+            self._write = self.__writeSerial
+
+        if (parallel):
```

Comment:
Not PEP8 (and not even consistent with the previous `if` statement format!).  Please re-read PEP8.